.github/workflows/holiday-generator/main.js

```diff
@@ -0,0 +1,125 @@
+import { createHash } from "node:crypto";
+import { writeFile } from "node:fs/promises";
+import { join } from "node:path";
+import { promisify } from "node:util";
+
+import Holidays from "date-holidays";
+import { createEvents as icsCreateEvents } from "ics";
+
+import { COUNTRIES, END_YEAR, FIXED_DATE_START_YEAR, ICS_PATH, SHOULD_LOG, START_YEAR, TYPE_WHITELIST } from "./config.js";
+
+// converting createEvents from ics from function with callback to async function for easier usage
+const createEvents = promisify(icsCreateEvents);
+
+/**
+ * Log info to console
+ * @param {*} toLog
+ */
+function log(toLog) {
+    if (SHOULD_LOG) {
+        console.log(toLog);
+    }
+}
+
+/**
+ * Get events for country given by code
+ * @param {string} countryCode
+ * @returns
+ */
+function getEvents(countryCode) {
+    const generator = new Holidays(countryCode);
+    const events = [];
+    for (let i = START_YEAR; i <= END_YEAR; i++) {
+        events.push(...generator.getHolidays(i).filter((x) => TYPE_WHITELIST.includes(x.type)));
+    }
+    return events;
+}
+
+/**
+ * Generates reproducible ID for holiday
+ * @param {string} countryCode
+ * @param {string} date
+ * @param {string} rule
+ * @returns
+ */
+function generateUid(countryCode, date, rule) {
+    const hashGen = createHash("sha256");
+    hashGen.update(`${countryCode},${date},${rule}`);
+    return hashGen.digest("hex");
+}
+
+/**
+ * Convert JS Date object to ics.DateTime array
+ * @param {Date} date
+ * @returns
+ */
+function getDateArray(date) {
+    return [date.getFullYear(), date.getMonth() + 1, date.getDate()];
+}
+
+/**
+ * Checks if holiday is a fixed-date holiday.
+ * Regex based on https://github.com/commenthol/date-holidays/blob/master/docs/specification.md#fixed-date
+ * @param {string} rule
+ * @returns
+ */
+function isFixedDate(rule) {
+    return /^\d\d-\d\d$/.test(rule);
+}
+
```

Comment:
This won't work for some fixed dates. The data that `getHolidays` returns doesn't strictly follow the `MM-DD` rule. For example, new year's rule:

![image](https://github.com/FossifyOrg/Calendar/assets/36371707/5e1ab0c2-8e3f-4ee5-ab01-d04b3f7fcf4f)

I think we can safely discard these conditions as long as the `MM-DD` pattern exists. There are more human-readable rules like `4th thursday in November` that we obviously can not parse without complicating things too much.

Reply:
Seeing the [specification](https://github.com/commenthol/date-holidays/blob/master/docs/specification.md#fixed-date), I'd change the regex to check if the pattern is `MM-DD` or `MM-DD and ...`. Too many different exceptions to handle, these two are the most straightforward.

Reply:
I 'think' we can get rid of the `and` too and match anything like `MM-DD.*`.

Reply:
In the USA's holidays, we have a rule `04-15 if friday then next monday if saturday,sunday then next tuesday`. If we'd simplify regex this way, sometimes date will be wrong.

Reply:
I couldn't get Google Calendar to show holidays in the future years but I see the start and end dates are different in date-holidays when the provided conditions are applicable. Let's keep `and` then, we'll worry about it after some feedback from users.